src/index.ts

```diff
@@ -7,6 +7,14 @@ import getRequestOptions from './get_request_options';
 
 export type ShouldRetry = (responseOrError: Response | Error) => boolean;
 
+export interface OnRequestEndOptions {
+  responseOrError: Response | Error;
+}
+
+export interface OnRequestOptions extends RequestInit {
+  url: string;
+  retryCount: number;
+}
```

Comment:
We could do the same thing for `OnRequestEndOptions` here:
```suggestion
export interface OnRequestOptions extends RequestInit {
  url: string;
  retryCount: number;
}

export interface OnRequestEndOptions extends OnRequestOptions {
  responseOrError: Response | Error;
}
```

and remove the `OnRequestOptions &` on line 30?